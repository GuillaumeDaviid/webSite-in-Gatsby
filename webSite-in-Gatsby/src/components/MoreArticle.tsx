import * as React from 'react'
import {useState} from 'react'
import './styles/MoreArticle.scss'
import { Link, graphql, StaticQuery } from 'gatsby'
import { GatsbyImage } from 'gatsby-plugin-image'

export default function MoreArticle () {
    return(
        <div className='MoreArticle' data-testid="article">
            <StaticQuery
      query={graphql`
      query {
        allMdx(sort: {fields: frontmatter___date, order: DESC}, limit: 4) {
          nodes {
            frontmatter {
              date(formatString: "MMMM D, YYYY")
              title
              slug
              featuredImage {
                childImageSharp {
                  gatsbyImageData(height: 100)
                }
              }
            }
            id
            excerpt
          }
        }
      }
      `}
      render={data => (
        <div>
          <div className='MoreArticle_items-elt'>{data.allMdx.nodes.map((node:any) => (
              <div className="MoreArticle_items-elt-item">
            <GatsbyImage image={node.frontmatter.featuredImage.childImageSharp.gatsbyImageData} alt='asdsf' className='blog_image'/>
            
            <h2 className='item_link'>
              <Link to={`/blog/${node.frontmatter.slug}`}>
                {node.frontmatter.title}
              </Link>
            </h2>
           
         </div>
        ))}
        </div>
        </div>
      )}
    />
        </div>
    )
}