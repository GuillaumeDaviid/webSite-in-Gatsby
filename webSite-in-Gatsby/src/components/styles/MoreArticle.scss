.MoreArticle {
    margin-bottom: 50px;
    &_items {
        &-elt {
            display: flex;
            justify-content: space-between;

            &-item {
                min-width: 24vw;
                display: flex;
                flex-direction: column;
                justify-content: center;
                align-items: center;
                text-align: center;
            }
        }

        
    }
}

@media (max-width: 750px) {
    .MoreArticle {
        &_items {
            &-elt {
                display: flex;
                flex-direction: column;
            }
        }
    }
}