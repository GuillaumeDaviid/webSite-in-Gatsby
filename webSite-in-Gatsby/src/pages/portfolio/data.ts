export const portfolioList = [
	{
        name: "Netflix",
		title: 'Netflix - Clone Netflix',
		description: "Réalisation d'un clone du site web de Netflix dans le cadre d'un projet personnel",
		id: '1',
		technique: `Projet réalisé avec le framework React.js et le préprocesseur Sass (SCSS). J'ai réalisé ce projet dans le cadre d'un travail personnel.`,
        task: `
         - Développement des fonctionnalités avec React.JS
         - Intégration responsive SASS (SCSS)
         - Axios / React-router / Material-UI
         - Test avec React-Testing Library et JEST
         - CI/CD avec Heroku et Github`,
		technologies: ["React.js", "Javascript", "SASS (SCSS)", "Axios", "React Router", "Material-UI", "React-Testing-Library / JEST", "Heroku"],
		link: "https://clone-guillaumedavid.herokuapp.com/",
        images : ['netflix'],
	},

    {
        name: "01 Academie",
		title: '01 Academie',
		description: "Création d'un site web de quizz qui permet de tester ses connaissances en programmation.",
		id: '2',
		technique: `Plusieurs version du projet, une avec du Javascript Vanilla et une autre avec React.JS. Backend réalisé en PHP.`,
        task:`
         - Intégration Responsive
         - Optimisation SEO
         - Développement de l'application avec React.JS 
         - Développement API REST avec PHP
         - Base de donnée SQL / MySQL
         - Tests unitaire JEST / React Testing Library`,
		technologies: ["React.js", "Javascript", "SASS (SCSS)", "Axios", "React Router", "PHP", "SQL / MySQL", "React-Testing-Library / JEST", "API REST", "OVH"],
		link: "https://github.com/GuillaumeDaviid/01-Academie-Version-React",
        images : ["01academie", "01courses", "01home"],
	},

    {
        name: "Jeu de Dame",
		title: 'Jeu de Dame',
		description: "Création d'un jeu de Dame dans le cadre d'un projet personnel.",
		id: '3',
		technique: `Plusieurs version du projet, une avec du Javascript Vanilla et une autre avec React.JS. Backend réalisé en PHP.`,
        task:`
         - Intégration Responsive
         - Optimisation SEO
         - Développement du jeu en Javascript Vanilla
         - Déploiement via Github
         - Test unitaire avec Jasmine`,
		technologies: ["Javascript", "HTML/CSS", "Github", "Jasmine"],
		link: "https://guillaumedaviid.github.io/Checkers-Game/",
        images : ["checkers"],
	},

    {
        name: "Commercia",
		title: 'Commercia - site e-commerce',
		description: "Création d'un e-commerce en React.js.",
		id: '4',
		technique: `Développement d'un site e-commerce en React avec Redux comme state manager.`,
        task:`
        - Développement de l'application en React.JS
        - Utilisation de Redux pour la gestion du State
        - Déploiement / Hébergement via Vercel (CI/CD)
        - Test Unitaire React-testing-library`,
		technologies: ["React.js", "Javascript", "SASS (SCSS)", "Redux", "React Router", "React-Testing-Library / JEST", "CI/CD", "Vercel"],
		link: "https://commercia.vercel.app/",
        images : ["commercia", "commercia-panier"],
	},

    {
        name: "Leagora",
		title: 'Leagora - Site de Visio conférence',
		description: "Je suis intervenu sur le projet en tant que développeur front-end.",
		id: '5',
		technique: `Entreprise qui commercialise une solution de visio-conférence, J'ai eu pour tache 
        d'ajouter différentes fonctionnalités à la visio en Javascript Vanilla basé sur la bibliothèque open source galene.`,
        task:`
        - Bouton de Kick et mute d'un utilisateur pour l'admin
        - Partage de fichier à tous les utilisateurs
        - Possibilité de changer ses périphériques
        - développement d'un chat privée`,
		technologies: ["Javascript", "HTML/CSS", "Galene", "GitLab", "Docker", "CI/CD"],
		link: "https://visio.leagora.fr/",
        images : ["leagora_black","leagora", "leagora-salle"],
	},
]

export default portfolioList