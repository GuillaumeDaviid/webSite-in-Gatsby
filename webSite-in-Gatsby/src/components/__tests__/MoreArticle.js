import React from "react"
import renderer from "react-test-renderer"
import MoreArticle from "../MoreArticle.tsx"
import { render, screen } from "@testing-library/react"
import { StaticQuery } from "gatsby"
import data from './data/MoreArticle.json'

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render(data)
  )
})

describe("MoreArticle", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<MoreArticle />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("should have a Article block", () => {
    render(<MoreArticle />)

    const article = screen.getByTestId('article');
    expect(article).toBeTruthy();
  })

  it("should have 4 items", () => {
    render(<MoreArticle />)

    const item = screen.getAllByTestId('items');
    expect(item).toHaveLength(4);
  })
})