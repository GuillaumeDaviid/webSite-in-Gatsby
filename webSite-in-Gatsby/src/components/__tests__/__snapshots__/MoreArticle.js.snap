// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div
  className="MoreArticle"
  data-testid="article"
>
  <div>
    <div
      className="MoreArticle_items-elt"
    >
      <div
        className="MoreArticle_items-elt-item"
      >
        <div
          className="gatsby-image-wrapper gatsby-image-wrapper-constrained blog_image"
          data-gatsby-image-wrapper=""
          style={
            {
              "backgroundColor": undefined,
              "display": "inline-block",
              "overflow": "hidden",
              "position": "relative",
              "verticalAlign": "top",
            }
          }
        >
          <div
            style={
              {
                "display": "block",
                "maxWidth": 167,
              }
            }
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;charset=utf-8,%3Csvg height='100' width='167' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
              style={
                {
                  "display": "block",
                  "maxWidth": "100%",
                  "position": "static",
                }
              }
            />
          </div>
          <div
            aria-hidden={true}
            data-placeholder-image=""
            style={
              {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              data-srcset="/static/7af5a20b055f7e5d1109e6aa6b2305c7/25c77/Nextjs.webp 42w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/1c99e/Nextjs.webp 84w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/e7b8f/Nextjs.webp 167w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/b19c2/Nextjs.webp 334w"
              sizes="(min-width: 167px) 167px, 100vw"
              type="image/webp"
            />
            <img
              alt="pourquoi-utiliser-nextjs"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/7af5a20b055f7e5d1109e6aa6b2305c7/577f0/Nextjs.png"
              data-srcset="/static/7af5a20b055f7e5d1109e6aa6b2305c7/f46b5/Nextjs.png 42w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/12d5d/Nextjs.png 84w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/577f0/Nextjs.png 167w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/f5f50/Nextjs.png 334w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 167px) 167px, 100vw"
              style={
                {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transform": "translateZ(0)",
                  "transition": "opacity 250ms linear",
                  "width": "100%",
                  "willChange": "opacity",
                }
              }
            />
          </picture>
          <noscript>
            <picture>
              <source
                sizes="(min-width: 167px) 167px, 100vw"
                srcSet="/static/7af5a20b055f7e5d1109e6aa6b2305c7/25c77/Nextjs.webp 42w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/1c99e/Nextjs.webp 84w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/e7b8f/Nextjs.webp 167w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/b19c2/Nextjs.webp 334w"
                type="image/webp"
              />
              <img
                alt="pourquoi-utiliser-nextjs"
                data-gatsby-image-ssr=""
                data-main-image=""
                decoding="async"
                loading="lazy"
                sizes="(min-width: 167px) 167px, 100vw"
                src="/static/7af5a20b055f7e5d1109e6aa6b2305c7/577f0/Nextjs.png"
                srcSet="/static/7af5a20b055f7e5d1109e6aa6b2305c7/f46b5/Nextjs.png 42w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/12d5d/Nextjs.png 84w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/577f0/Nextjs.png 167w,/static/7af5a20b055f7e5d1109e6aa6b2305c7/f5f50/Nextjs.png 334w"
                style={
                  {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "transition": "opacity 250ms linear",
                    "width": "100%",
                    "willChange": "opacity",
                  }
                }
              />
            </picture>
          </noscript>
          <script
            dangerouslySetInnerHTML={
              {
                "__html": "const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}",
              }
            }
            type="module"
          />
        </div>
        <h2
          className="item_link"
          data-testid="items"
        >
          <a
            href="/blog/pourquoi-utiliser-nextjs"
          >
            Pourquoi utiliser Next.JS ?
          </a>
        </h2>
      </div>
      <div
        className="MoreArticle_items-elt-item"
      >
        <div
          className="gatsby-image-wrapper gatsby-image-wrapper-constrained blog_image"
          data-gatsby-image-wrapper=""
          style={
            {
              "backgroundColor": undefined,
              "display": "inline-block",
              "overflow": "hidden",
              "position": "relative",
              "verticalAlign": "top",
            }
          }
        >
          <div
            style={
              {
                "display": "block",
                "maxWidth": 178,
              }
            }
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;charset=utf-8,%3Csvg height='100' width='178' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
              style={
                {
                  "display": "block",
                  "maxWidth": "100%",
                  "position": "static",
                }
              }
            />
          </div>
          <div
            aria-hidden={true}
            data-placeholder-image=""
            style={
              {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              data-srcset="/static/2bab627a79337d768c8221c571d2c81a/1b4b4/gatsby.webp 45w,/static/2bab627a79337d768c8221c571d2c81a/4c978/gatsby.webp 89w,/static/2bab627a79337d768c8221c571d2c81a/744a3/gatsby.webp 178w,/static/2bab627a79337d768c8221c571d2c81a/723ea/gatsby.webp 356w"
              sizes="(min-width: 178px) 178px, 100vw"
              type="image/webp"
            />
            <img
              alt="pourquoi-utiliser-gatsby-pour-votre-site-web"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/2bab627a79337d768c8221c571d2c81a/ccf6a/gatsby.png"
              data-srcset="/static/2bab627a79337d768c8221c571d2c81a/1170e/gatsby.png 45w,/static/2bab627a79337d768c8221c571d2c81a/b2b7e/gatsby.png 89w,/static/2bab627a79337d768c8221c571d2c81a/ccf6a/gatsby.png 178w,/static/2bab627a79337d768c8221c571d2c81a/3c234/gatsby.png 356w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 178px) 178px, 100vw"
              style={
                {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transform": "translateZ(0)",
                  "transition": "opacity 250ms linear",
                  "width": "100%",
                  "willChange": "opacity",
                }
              }
            />
          </picture>
          <noscript>
            <picture>
              <source
                sizes="(min-width: 178px) 178px, 100vw"
                srcSet="/static/2bab627a79337d768c8221c571d2c81a/1b4b4/gatsby.webp 45w,/static/2bab627a79337d768c8221c571d2c81a/4c978/gatsby.webp 89w,/static/2bab627a79337d768c8221c571d2c81a/744a3/gatsby.webp 178w,/static/2bab627a79337d768c8221c571d2c81a/723ea/gatsby.webp 356w"
                type="image/webp"
              />
              <img
                alt="pourquoi-utiliser-gatsby-pour-votre-site-web"
                data-gatsby-image-ssr=""
                data-main-image=""
                decoding="async"
                loading="lazy"
                sizes="(min-width: 178px) 178px, 100vw"
                src="/static/2bab627a79337d768c8221c571d2c81a/ccf6a/gatsby.png"
                srcSet="/static/2bab627a79337d768c8221c571d2c81a/1170e/gatsby.png 45w,/static/2bab627a79337d768c8221c571d2c81a/b2b7e/gatsby.png 89w,/static/2bab627a79337d768c8221c571d2c81a/ccf6a/gatsby.png 178w,/static/2bab627a79337d768c8221c571d2c81a/3c234/gatsby.png 356w"
                style={
                  {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "transition": "opacity 250ms linear",
                    "width": "100%",
                    "willChange": "opacity",
                  }
                }
              />
            </picture>
          </noscript>
          <script
            dangerouslySetInnerHTML={
              {
                "__html": "const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}",
              }
            }
            type="module"
          />
        </div>
        <h2
          className="item_link"
          data-testid="items"
        >
          <a
            href="/blog/pourquoi-utiliser-gatsby-pour-votre-site-web"
          >
            Pourquoi utiliser Gatsby pour votre site web ?
          </a>
        </h2>
      </div>
      <div
        className="MoreArticle_items-elt-item"
      >
        <div
          className="gatsby-image-wrapper gatsby-image-wrapper-constrained blog_image"
          data-gatsby-image-wrapper=""
          style={
            {
              "backgroundColor": undefined,
              "display": "inline-block",
              "overflow": "hidden",
              "position": "relative",
              "verticalAlign": "top",
            }
          }
        >
          <div
            style={
              {
                "display": "block",
                "maxWidth": 172,
              }
            }
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;charset=utf-8,%3Csvg height='100' width='172' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
              style={
                {
                  "display": "block",
                  "maxWidth": "100%",
                  "position": "static",
                }
              }
            />
          </div>
          <div
            aria-hidden={true}
            data-placeholder-image=""
            style={
              {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              data-srcset="/static/214283de647934d3bd26462e8911fd90/3132b/skills.webp 43w,/static/214283de647934d3bd26462e8911fd90/92fbd/skills.webp 86w,/static/214283de647934d3bd26462e8911fd90/f3a54/skills.webp 172w,/static/214283de647934d3bd26462e8911fd90/0cff0/skills.webp 344w"
              sizes="(min-width: 172px) 172px, 100vw"
              type="image/webp"
            />
            <img
              alt="les-competences-indispensables-pour-un-developpeur"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/214283de647934d3bd26462e8911fd90/cca54/skills.jpg"
              data-srcset="/static/214283de647934d3bd26462e8911fd90/e102b/skills.jpg 43w,/static/214283de647934d3bd26462e8911fd90/3ee87/skills.jpg 86w,/static/214283de647934d3bd26462e8911fd90/cca54/skills.jpg 172w,/static/214283de647934d3bd26462e8911fd90/4db5e/skills.jpg 344w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 172px) 172px, 100vw"
              style={
                {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transform": "translateZ(0)",
                  "transition": "opacity 250ms linear",
                  "width": "100%",
                  "willChange": "opacity",
                }
              }
            />
          </picture>
          <noscript>
            <picture>
              <source
                sizes="(min-width: 172px) 172px, 100vw"
                srcSet="/static/214283de647934d3bd26462e8911fd90/3132b/skills.webp 43w,/static/214283de647934d3bd26462e8911fd90/92fbd/skills.webp 86w,/static/214283de647934d3bd26462e8911fd90/f3a54/skills.webp 172w,/static/214283de647934d3bd26462e8911fd90/0cff0/skills.webp 344w"
                type="image/webp"
              />
              <img
                alt="les-competences-indispensables-pour-un-developpeur"
                data-gatsby-image-ssr=""
                data-main-image=""
                decoding="async"
                loading="lazy"
                sizes="(min-width: 172px) 172px, 100vw"
                src="/static/214283de647934d3bd26462e8911fd90/cca54/skills.jpg"
                srcSet="/static/214283de647934d3bd26462e8911fd90/e102b/skills.jpg 43w,/static/214283de647934d3bd26462e8911fd90/3ee87/skills.jpg 86w,/static/214283de647934d3bd26462e8911fd90/cca54/skills.jpg 172w,/static/214283de647934d3bd26462e8911fd90/4db5e/skills.jpg 344w"
                style={
                  {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "transition": "opacity 250ms linear",
                    "width": "100%",
                    "willChange": "opacity",
                  }
                }
              />
            </picture>
          </noscript>
          <script
            dangerouslySetInnerHTML={
              {
                "__html": "const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}",
              }
            }
            type="module"
          />
        </div>
        <h2
          className="item_link"
          data-testid="items"
        >
          <a
            href="/blog/les-competences-indispensables-pour-un-developpeur"
          >
            Les compétences indispensables pour un développeur !
          </a>
        </h2>
      </div>
      <div
        className="MoreArticle_items-elt-item"
      >
        <div
          className="gatsby-image-wrapper gatsby-image-wrapper-constrained blog_image"
          data-gatsby-image-wrapper=""
          style={
            {
              "backgroundColor": undefined,
              "display": "inline-block",
              "overflow": "hidden",
              "position": "relative",
              "verticalAlign": "top",
            }
          }
        >
          <div
            style={
              {
                "display": "block",
                "maxWidth": 183,
              }
            }
          >
            <img
              alt=""
              aria-hidden="true"
              role="presentation"
              src="data:image/svg+xml;charset=utf-8,%3Csvg height='100' width='183' xmlns='http://www.w3.org/2000/svg' version='1.1'%3E%3C/svg%3E"
              style={
                {
                  "display": "block",
                  "maxWidth": "100%",
                  "position": "static",
                }
              }
            />
          </div>
          <div
            aria-hidden={true}
            data-placeholder-image=""
            style={
              {
                "height": "100%",
                "left": 0,
                "position": "absolute",
                "top": 0,
                "width": "100%",
              }
            }
          />
          <picture>
            <source
              data-srcset="/static/5dd72b61e716eae08bc0938e7b9d38f1/2b1aa/React.webp 46w,/static/5dd72b61e716eae08bc0938e7b9d38f1/5810e/React.webp 92w,/static/5dd72b61e716eae08bc0938e7b9d38f1/4c67c/React.webp 183w,/static/5dd72b61e716eae08bc0938e7b9d38f1/0a94f/React.webp 366w"
              sizes="(min-width: 183px) 183px, 100vw"
              type="image/webp"
            />
            <img
              alt="5-bibliotheques-react-a-utiliser"
              data-gatsby-image-ssr=""
              data-main-image=""
              data-src="/static/5dd72b61e716eae08bc0938e7b9d38f1/f39b0/React.jpg"
              data-srcset="/static/5dd72b61e716eae08bc0938e7b9d38f1/06f15/React.jpg 46w,/static/5dd72b61e716eae08bc0938e7b9d38f1/4e3bd/React.jpg 92w,/static/5dd72b61e716eae08bc0938e7b9d38f1/f39b0/React.jpg 183w,/static/5dd72b61e716eae08bc0938e7b9d38f1/a1cd5/React.jpg 366w"
              decoding="async"
              loading="lazy"
              sizes="(min-width: 183px) 183px, 100vw"
              style={
                {
                  "backgroundColor": undefined,
                  "height": "100%",
                  "left": 0,
                  "objectFit": undefined,
                  "objectPosition": undefined,
                  "opacity": 0,
                  "position": "absolute",
                  "top": 0,
                  "transform": "translateZ(0)",
                  "transition": "opacity 250ms linear",
                  "width": "100%",
                  "willChange": "opacity",
                }
              }
            />
          </picture>
          <noscript>
            <picture>
              <source
                sizes="(min-width: 183px) 183px, 100vw"
                srcSet="/static/5dd72b61e716eae08bc0938e7b9d38f1/2b1aa/React.webp 46w,/static/5dd72b61e716eae08bc0938e7b9d38f1/5810e/React.webp 92w,/static/5dd72b61e716eae08bc0938e7b9d38f1/4c67c/React.webp 183w,/static/5dd72b61e716eae08bc0938e7b9d38f1/0a94f/React.webp 366w"
                type="image/webp"
              />
              <img
                alt="5-bibliotheques-react-a-utiliser"
                data-gatsby-image-ssr=""
                data-main-image=""
                decoding="async"
                loading="lazy"
                sizes="(min-width: 183px) 183px, 100vw"
                src="/static/5dd72b61e716eae08bc0938e7b9d38f1/f39b0/React.jpg"
                srcSet="/static/5dd72b61e716eae08bc0938e7b9d38f1/06f15/React.jpg 46w,/static/5dd72b61e716eae08bc0938e7b9d38f1/4e3bd/React.jpg 92w,/static/5dd72b61e716eae08bc0938e7b9d38f1/f39b0/React.jpg 183w,/static/5dd72b61e716eae08bc0938e7b9d38f1/a1cd5/React.jpg 366w"
                style={
                  {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "transition": "opacity 250ms linear",
                    "width": "100%",
                    "willChange": "opacity",
                  }
                }
              />
            </picture>
          </noscript>
          <script
            dangerouslySetInnerHTML={
              {
                "__html": "const t="undefined"!=typeof HTMLImageElement&&"loading"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll("img[data-main-image]");for(let e of t){e.dataset.src&&(e.setAttribute("src",e.dataset.src),e.removeAttribute("data-src")),e.dataset.srcset&&(e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset"));const t=e.parentNode.querySelectorAll("source[data-srcset]");for(let e of t)e.setAttribute("srcset",e.dataset.srcset),e.removeAttribute("data-srcset");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector("[data-placeholder-image]").style.opacity=0)}}",
              }
            }
            type="module"
          />
        </div>
        <h2
          className="item_link"
          data-testid="items"
        >
          <a
            href="/blog/5-bibliotheques-react-a-utiliser"
          >
            5 Bibliothèques React à utiliser !
          </a>
        </h2>
      </div>
    </div>
  </div>
</div>
`;
